nixos: remove optional builtins prefixes from prelude functions

trueNAHO opened the pull request:
```
Remove optional builtins prefixes from prelude functions by running:

    builtins=(
      abort
      baseNameOf
      break
      derivation
      derivationStrict
      dirOf
      false
      fetchGit
      fetchMercurial
      fetchTarball
      fetchTree
      fromTOML
      import
      isNull
      map
      null
      placeholder
      removeAttrs
      scopedImport
      throw
      toString
      true
    )

    fd \
      --exclude doc/manual/release-notes \
      --type file \
      . \
      nixos \
      --exec-batch sed --in-place --regexp-extended "
        s/\<builtins\.($(
          printf '%s\n' "${builtins[@]}" |
            paste --delimiter '|' --serial -
        ))\>/\1/g
      "

    nix fmt
```

Related PRs intentionally split up to simplify review and merging:

1. https://github.com/NixOS/nixpkgs/pull/447401
2. https://github.com/NixOS/nixpkgs/pull/447402
3. https://github.com/NixOS/nixpkgs/pull/447404
4. https://github.com/NixOS/nixpkgs/pull/447403
5. https://github.com/NixOS/nixpkgs/pull/444432

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [X] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

qweered reviewed (approved):
Lgtm

trueNAHO commented on nixos/doc/manual/release-notes/rl-2411.section.md:
```diff
@@ -1042,9 +1042,9 @@ New and extended interfaces:
 - Various builtins are now reexported in a more standard way:
-  - `lib.map` -> `lib.lists.map` -> `builtins.map`
+  - `lib.map` -> `lib.lists.map` -> `map`
   - `lib.intersectAttrs` -> `lib.attrsets.intersectAttrs` -> `builtins.intersectAttrs`
-  - `lib.removeAttrs` -> `lib.attrsets.removeAttrs` -> `builtins.removeAttrs`
+  - `lib.removeAttrs` -> `lib.attrsets.removeAttrs` -> `removeAttrs`
```
Should we revert those changes or extend them:

```diff
-  - `lib.map` -> `lib.lists.map` -> `builtins.map`
+  - `lib.map` -> `lib.lists.map` -> `builtins.map` -> `map`
   - `lib.intersectAttrs` -> `lib.attrsets.intersectAttrs` -> `builtins.intersectAttrs`
-  - `lib.removeAttrs` -> `lib.attrsets.removeAttrs` -> `builtins.removeAttrs`
+  - `lib.removeAttrs` -> `lib.attrsets.removeAttrs` -> `builtins.removeAttrs` -> `removeAttrs`
```

MattSturgeon replied:
I don't think we should be editing the 24.11 release notes.

MattSturgeon reviewed (approved):
I skimmed through the diff and couldn't see any issues. I've **not** attempted to reproduce via the script.

Other than merge conflicts and waiting a little for anyone who wants to review more thoroughly, this LGTM.

qweered reviewed (approved):
Skimmed through changes, nothing questionable

trueNAHO committed:
nixos: remove optional builtins prefixes from prelude functions

Remove optional builtins prefixes from prelude functions by running:

    builtins=(
      abort
      baseNameOf
      break
      derivation
      derivationStrict
      dirOf
      false
      fetchGit
      fetchMercurial
      fetchTarball
      fetchTree
      fromTOML
      import
      isNull
      map
      null
      placeholder
      removeAttrs
      scopedImport
      throw
      toString
      true
    )

    fd \
      --exclude doc/manual/release-notes \
      --type file \
      . \
      nixos \
      --exec-batch sed --in-place --regexp-extended "
        s/\<builtins\.($(
          printf '%s\n' "${builtins[@]}" |
            paste --delimiter '|' --serial -
        ))\>/\1/g
      "

    nix fmt

MattSturgeon commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 447403`
Commit: [`a2ed7e8d882c2b9deba7b7c2cc8e1a97b29aa415`](https://github.com/NixOS/nixpkgs/commit/a2ed7e8d882c2b9deba7b7c2cc8e1a97b29aa415) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/a2ed7e8d882c2b9deba7b7c2cc8e1a97b29aa415..pull/447403/head))
Merge: [`b38605475a2a014af134ffbb36895fa14cecafdd`](https://github.com/NixOS/nixpkgs/commit/b38605475a2a014af134ffbb36895fa14cecafdd)

Logs: https://github.com/MattSturgeon/nixpkgs-review-gha/actions/runs/21037047475


---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:fast_forward: 2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>

MattSturgeon reviewed (approved):
I skimmed the diff _very_ quickly again. No obvious issues.

Rebuild count isn't particularly relevant to NixOS modules, but only seeing two is still encouraging:

- [nixos-install-tools](https://search.nixos.org/packages?channel=unstable&show=nixos-install-tools&from=0&size=50&sort=relevance&type=packages&query=nixos-install-tools)
- [tests.nixos-functions.nixos-test](https://search.nixos.org/packages?channel=unstable&show=tests.nixos-functions.nixos-test&from=0&size=50&sort=relevance&type=packages&query=tests.nixos-functions.nixos-test)

I'll [run](https://github.com/MattSturgeon/nixpkgs-review-gha/actions/runs/21037047475) a nixpkgs-review just to be safe (not sure if nixpkgs-review-gha will be able to build `tests.nixos-functions.nixos-test`, but we'll see). EDIT: ah, it's blacklisted.

I aim to get this merged promptly before the inevitable conflicts reappear. If issues are found post-merge we can fix or revert as necessary.

bjornfor commented:
@trueNAHO: Can you create a new PR for the release-25.11 branch, please? (I presume since it's auto-generated that it might be easier to do it that way than to cherry-pick and resolve conflicts.)

trueNAHO commented:
> @trueNAHO: Can you create a new PR for the release-25.11 branch, please? (I presume since it's auto-generated that it might be easier to do it that way than to cherry-pick and resolve conflicts.)

I opened https://github.com/NixOS/nixpkgs/pull/480781.

> > This needs to be backported or else other backports are going to be difficult.
>
> Good point. This applies equally to the other PRs in this series:
>
> 1. https://github.com/NixOS/nixpkgs/pull/447401
> 2. https://github.com/NixOS/nixpkgs/pull/447402
> 3. https://github.com/NixOS/nixpkgs/pull/447404
> 4. https://github.com/NixOS/nixpkgs/pull/444432
>
> We asked for the initial PRs to be split for easier review, however I'm fine to review a single backport.

Based on the following output, the other PRs were already merged prior to the `release-25.11` branch-off, meaning only this PR needs to be backported to `release-25.11`:

```bash
$ for commit in \
      b50da098117870e48cba6c110d290cb2a2dadf93 \
      44998896c6a811fcc4021e8ca75243b5d53e605b \
      6d1d15bb3610a98c8d6ba63e7e792cec53215227 \
      c078ee646535e63a08f709d3f1b5d122407e6f12 \
      d9335d5dd7057963bf0006432427e115b2825c4d
  do
      git merge-base --is-ancestor "$commit" release-25.11 || echo "$commit"
  done
c078ee646535e63a08f709d3f1b5d122407e6f12
```

> > Can you create a new PR for the release-25.11 branch, please? (I presume since it's auto-generated that it might be easier to do it that way than to cherry-pick and resolve conflicts.)
>
> If the backport is made of 5 commits (freshly generated, but still tagged with `(cherry picked from commit ...)`), then CI will be able to higlight the difference with the original commits from master, aiding review.

Are you referring to some general GitHub feature, or just https://github.com/NixOS/nixpkgs/pull/480781#pullrequestreview-3672235520? I usually run something like the following locally:

```fish
diff <(git show "$COMMIT_1") <(git show "$COMMIT_2")
```
